### PR TITLE
feat: Enable submission title modification

### LINKED ISSUE #1735: Submission Title Modification
### Problem/Why
After a submission is uploaded/created, there is no way in the application for a user to modify the submission's title. This can be an issue for those who wish to correct mistakes in or improve a submission's title.

### Goal/What
Enable functionality that allows a user with sufficient privilege (submitter, review coordinator, editor, or administrator) to modify a submission's title.

### Solution/How
Update `SubmissionPolicy.php` for updating the submission's title so that only the intended roles are allowed to modify the title. Update the GraphQL endpoint to use `SubmissionPolicy.php` for updating a submission title's. Modify and add functionality to the Vue template for the Submission Details page to use this updated endpoint for modifying a submission's title. Add validation on the backend and frontend to ensure that the title is a maximum of 512 characters and is not empty. 

### To Do / Acceptance Criteria

- [x] Update `SubmissionPolicy.php` and utilize this policy in the GraphQL endpoint
- [x] Update the Submission Details Vue template with necessary components and functionality
- [x] Add validation
- [x] Add PHPUnit testing to ensure API works as expected
- [x] Add Cypress testing to ensure feature works as expected

### Review Checklist

- [ ] A submission's title can be modified by a submitter of the submission
- [ ] A submission's title cannot be modified by a reviewer of the submission

### Accessibility Checklist

- [ ] Modifying the title can be done solely with a keyboard

### PR DESCRIPTION
**Pull request by gmeben**
This pull request:

* Allows a user to modify submission titles on the Submission Details page
    * Allowed users: submitters of submissions, review coordinators, editors, and admins
* Adds validation for the mutation that updates submission
* Extends `SubmissionPolicy.php` for title modification to enact authorization rules by role
* Updates styles
    * Makes the color for textarea placeholder text consistent with the color for text input placeholder text
    * Fixes the failing color contrast for tooltips so that it passes WCAG AA ratio
* Fixes console warnings on the Submission Details page 
    * Caused by i18n template strings for notifications and audit logs: `[intlify] Not found parent scope. use the global scope.`
    * Caused by a lack of specified response `id`s for publications of queried users
* Improves the clarity of the Activity section on the Submission Details page so that title updates can be easily distinguished from status updates
* Adds PHPUnit testing for submissions
    * Extends `TestCast.php` for creating generic users by role with submissions
    * Fixes role references on lines 899–904 in `SubmissionTest.php`
* Adds a Cypress test to the Submission Details page

Closes #1735

### CODE REVIEW

File: backend/graphql/submission.graphql [hunk 1]
```diff
@@ -147,7 +148,7 @@ input AcceptSubmissionInviteInput @validator(class: "SubmissionInvitationAccepta
     password: String!
 }
 
-input UpdateSubmissionInput {
+input UpdateSubmissionInput @validator(class: "SubmissionUpdateInputValidator") {
```
Comment (wreality) [inline backend/graphql/submission.graphql:151]:
```suggestion
input UpdateSubmissionInput @validator {
```
*if* you rename the validator class to `UpdateSubmissionInputValidator` you won't need to specify the validator class as it'll follow convention.

File: backend/app/GraphQL/Validators/SubmissionUpdateInputValidator.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace App\GraphQL\Validators;
+
+use Nuwave\Lighthouse\Validation\Validator;
+
+class SubmissionUpdateInputValidator extends Validator
```
Comment (wreality) [inline backend/app/GraphQL/Validators/SubmissionUpdateInputValidator.php:8]: Rename this file and class to UpdateSubmissionInputValidator and it'll follow the convention established by lighthouse.

**Review by wreality — APPROVED** [review]
Only two inter-related suggestions.

File: backend/graphql/submission.graphql (hunk 1 above)
Comment (gmeben) [reply backend/graphql/submission.graphql:151]: Interesting!

Comment (gmeben) [reply backend/graphql/submission.graphql]: Refactored as suggested

File: backend/app/GraphQL/Validators/SubmissionUpdateInputValidator.php (hunk 2 above)
Comment (gmeben) [reply backend/app/GraphQL/Validators/SubmissionUpdateInputValidator.php]: Refactored as suggested

**Pull request merged**